js/huobi.js

```diff
@@ -4698,39 +4727,31 @@ module.exports = class huobi extends Exchange {
-        const networks = this.safeValue (this.options, 'networks', {});
-        const network = this.safeStringUpper (networks, rawNetwork, rawNetwork);
-        params = this.omit (params, 'network');
-        const response = await this.fetchDepositAddressesByNetwork (code, params);
-        let result = undefined;
-        if (network === undefined) {
-            result = this.safeValue (response, code);
-            if (result === undefined) {
-                const alias = this.safeString (networks, code, code);
-                result = this.safeValue (response, alias);
-                if (result === undefined) {
-                    const defaultNetwork = this.safeString (this.options, 'defaultNetwork', 'ERC20');
-                    result = this.safeValue (response, defaultNetwork);
-                    if (result === undefined) {
-                        const values = Object.values (response);
-                        result = this.safeValue (values, 0);
-                        if (result === undefined) {
-                            throw new InvalidAddress (this.id + ' fetchDepositAddress() cannot find deposit address for ' + code);
-                        }
-                    }
-                }
-            }
-            return result;
-        }
-        result = this.safeValue (response, network);
-        if (result === undefined) {
-            throw new InvalidAddress (this.id + ' fetchDepositAddress() cannot find ' + network + ' deposit address for ' + code);
-        }
-        return result;
+        const [ networkCode, paramsOmited ] = this.handleNetworkCodeAndParams (params);
+        const indexedAddresses = await this.fetchDepositAddressesByNetwork (code, paramsOmited);
+        const selectedNetworkId = this.selectNetworkIdFromAvailableNetworks (code, networkCode, indexedAddresses);
+        return indexedAddresses[selectedNetworkId];
+    }
+
+    networkCodeFromPairId (info, key, currencyCode) {
+        const networkPairId = this.safeString (info, key);
+        const networkIds = this.safeValue (this.networkIdByNetworkPairId, currencyCode, {});
+        const networkId = this.safeValue (networkIds, networkPairId, networkPairId);
+        return this.networkIdToCode (networkId);
     }
 
-    async fetchWithdrawAddressesByNetwork (code, params = {}) {
+    selectNetworkIdFromAvailableNetworksCustom (currencyCode, networkCode, networkEntriesIndexed) {
+        const renamedIndexedNetworkIds = {};
+        const networkPairIdKeys = Object.keys (networkEntriesIndexed);
+        for (let i = 0; i < networkPairIdKeys.length; i++) {
+            const pairId = networkPairIdKeys[i];
+            const networkId = this.networkIdByNetworkPairId[currencyCode][pairId];
+            renamedIndexedNetworkIds[networkId] = networkEntriesIndexed[pairId];
+        }
+        return renamedIndexedNetworkIds;
+    }
+
+    async fetchWithdrawAddresses (code, note = undefined, networkCode = undefined, params = {}) {
```

Comment:
There shouldn't be `fetchWithdrawalAddress` and `fetchWithdrawalAddressesByNetwork`. They seem to be just a copy-pasted version for `fetchDeposit...` counterparts, however, they are not compliant to withdrawals and cant be used here. reason is that, unlike deposit address (where you can only have 1 deposit address per currency-network), for withdrawals you can have different recipients / many addresses for same coin (i.e. `USDT`) on same network (i.e. `TRC20`) , so it's not possible to fetch single withdrawal address by just `currencyCode` and `networkCode`. 
I have also thought to fetch them by 'note' (which seems more unique) but not really unique, as you might also have same title/note for different withdrawal addresses.
So, fetching **singular** withdrawal address by anything seems non-viable. The only unique thing that you can match to singular address-object , is itself "withdrawal address" (`0x9eFF....`), so it has no point to fetch withdrawal address by "address" itself :)) 

Thus, I think that the only optimal way is to have `fetchWithdrawalAddresses` (plural) and we can have two arguments `note` and `networkCode` to filter out the response, but the response should be array(list), not a dictionary. that seems the only good path for ccxt.